shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -655,7 +659,7 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
    */
   @VisibleForTesting
   public void registerOnBackInvokedCallback() {
-    if (Build.VERSION.SDK_INT >= 33) {
+    if (Build.VERSION.SDK_INT >= 33 && onBackInvokedCallback != null) {
```

Comment:
Do we need both of these checks?

Reply:
Unfortunately, yes, since the compiler does not follow what's going on inside the `createOnBackInvokedCallback` method.

I can remove the method and use a nested Elvis operator:
```java
  final OnBackInvokedCallback onBackInvokedCallback =
          Build.VERSION.SDK_INT == 33 ? this::onBackPressed : (Build.VERSION.SDK_INT >= 34 ?
                  new OnBackAnimationCallback() {
                    @Override
                    public void onBackInvoked() {
                      commitBackGesture();
                    }

                    @Override
                    public void onBackCancelled() {
                      cancelBackGesture();
                    }

                    @Override
                    public void onBackProgressed(@NonNull BackEvent backEvent) {
                      updateBackGestureProgress(backEvent);
                    }

                    @Override
                    public void onBackStarted(@NonNull BackEvent backEvent) {
                      startBackGesture(backEvent);
                    }
                  } : null);
```

Or mark `createOnBackInvokedCallback` return value as `@NonNull`:
```java
  final OnBackInvokedCallback onBackInvokedCallback = Build.VERSION.SDK_INT < 33 ? null : createOnBackInvokedCallback();

  @NonNull
  private OnBackInvokedCallback createOnBackInvokedCallback() {
    if (Build.VERSION.SDK_INT >= 34) {
      return new OnBackAnimationCallback() {
        @Override
        public void onBackInvoked() {
          commitBackGesture();
        }

        @Override
        public void onBackCancelled() {
          cancelBackGesture();
        }

        @Override
        public void onBackProgressed(@NonNull BackEvent backEvent) {
          updateBackGestureProgress(backEvent);
        }

        @Override
        public void onBackStarted(@NonNull BackEvent backEvent) {
          startBackGesture(backEvent);
        }
      };
    }

    return this::onBackPressed;
  }
```